internal/templates/image-based-install/template.go

```diff
@@ -0,0 +1,178 @@
+metadata:
+  annotations:
+    metaclusterinstall.openshift.io/sync-wave: "2"
+  labels:
+    installer.name: multiclusterhub
+    installer.namespace: open-cluster-management
+  name: "{{ .Site.ClusterName }}"
+  namespace: "{{ .Site.ClusterName }}"
+spec:
+  clusterName: "{{ .Site.ClusterName }}"
+  clusterNamespace: "{{ .Site.ClusterName }}"
+  clusterLabels:
+    cloud: auto-detect
+    vendor: auto-detect
+  applicationManager:
+    enabled: true
+  certPolicyController:
+    enabled: true
+  iamPolicyController:
+    enabled: true
+  policyController:
+    enabled: true
+  searchCollector:
+    enabled: false`
+
+const ManagedCluster = `apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  name: "{{ .Site.ClusterName }}"
+  labels:
+{{ .Site.ClusterLabels | toYaml | indent 4 }}
+  annotations:
+    metaclusterinstall.openshift.io/sync-wave: "2"
+spec:
+  hubAcceptsClient: true`
+
+const BareMetalHost = `apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: "{{ .SpecialVars.CurrentNode.HostName }}"
+  namespace: "{{ .Site.ClusterName }}"
+  annotations:
+    metaclusterinstall.openshift.io/sync-wave: "1"
+    inspect.metal3.io: "{{ .SpecialVars.CurrentNode.IronicInspect }}"
+{{ if .SpecialVars.CurrentNode.NodeLabels }}
+    bmac.agent-install.openshift.io.node-label:
+{{ .SpecialVars.CurrentNode.NodeLabels | toYaml | indent 6 }}
+{{ end }}
+    bmac.agent-install.openshift.io/hostname: "{{ .SpecialVars.CurrentNode.HostName }}"
+{{ if .SpecialVars.CurrentNode.InstallerArgs  }}
+    bmac.agent-install.openshift.io/installer-args: {{ .SpecialVars.CurrentNode.InstallerArgs  }}
+{{ end }}
+{{ if .SpecialVars.CurrentNode.IgnitionConfigOverride }}
+    bmac.agent-install.openshift.io/ignition-config-overrides: {{ .SpecialVars.CurrentNode.IgnitionConfigOverride }}
+{{ end }}
+    bmac.agent-install.openshift.io/role: "{{ .SpecialVars.CurrentNode.Role }}"
+  labels:
+    infraenvs.agent-install.openshift.io: "{{ .Site.ClusterName }}"
+spec:
+  bootMode: "{{ .SpecialVars.CurrentNode.BootMode }}"
+  bmc:
+    address: "{{ .SpecialVars.CurrentNode.BmcAddress }}"
+    disableCertificateVerification: true
+    credentialsName: "{{ .SpecialVars.CurrentNode.BmcCredentialsName.Name }}"
+  bootMACAddress: "{{ .SpecialVars.CurrentNode.BootMACAddress }}"
+  automatedCleaningMode: "{{ .SpecialVars.CurrentNode.AutomatedCleaningMode }}"
+  online: true
+{{ if (anyFieldDefined .SpecialVars.CurrentNode.RootDeviceHints) }}
+  rootDeviceHints:
+{{ .SpecialVars.CurrentNode.RootDeviceHints | toYaml | indent 4 }}
+{{ end }}`
+
+func GetClusterTemplates() map[string]string {
+	data := make(map[string]string)
+	data["ClusterDeployment"] = ClusterDeployment
+	data["ManagedCluster"] = ManagedCluster
+	data["KlusterletAddonConfig"] = KlusterletAddonConfig
+	return data
+}
+
+func GetNodeTemplates() map[string]string {
+	data := make(map[string]string)
+	data["ImageClusterInstall"] = ImageClusterInstall
```

Comment:
I'm not sure what's the meaning of the division between cluster and node templates. 
Although ImageClusterInstall currently (and for the foreseeable future) only supports single node installation I think that the ImageClusterInstall resource belongs to the clusterTemplates.

Reply:
The current `v1alpha1` version of the `ImageClusterInstall` manifest references node-specific fields such as the hostname and BMH references, in addition to network-related configs pertaining to the node. As per the design of the cluster and node templates for the SiteConfig API, a cluster template should only reference cluster (site-wide) fields, whereas, a node template may reference both cluster and node-specific fields. For example, the `AgentClusterInstall` CR only refers to cluster-wide fields, thus, it is categorized as a cluster template.